sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -376,6 +381,21 @@ def __init__(
     def __len__(self):
         return self.size
 
+    def _get_encryption_data_request(self):
+        # Save current request cls
+        download_cls = self._request_options.pop('cls', None)
+        # Adjust cls for get_properties
+        self._request_options['cls'] = deserialize_blob_properties
+
+        properties = self._clients.blob.get_properties(**self._request_options)
```

Comment:
Looks like we decided on the pre-emptive `get_properties` approach - I'm happy so long as non-encryption path is unaffected. As we discussed - I don't think much sleep will be lost if the client-side encryption path is not so performant.

Reply:
Yep, felt like the easiest/least error-prone approach. We can always revisit if this starts to cause a problem, but I think this should be acceptable. And yep, this will only be triggered for requests where the user has provided encryption info for decryption. There is of course the chance the blob is not actually encrypted in which case this is a useless call, but I think that's also justifiable if encryption information is provided.

The one note is this applies to V1 now as well which is a bit of a change in behavior/performance of V1. I don't think this should be a problem since we are deprecated V1 but let me know if you feel otherwise.